app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -153,15 +152,12 @@ public void render(Chart chart, Map<String, Obs> latestObservations,
                 }
             }
             addObservations(observations);
+            insertEmptyColumns();
         }
 
         void addObservations(List<Obs> observations) {
             for (Obs obs : observations) {
                 if (obs == null) continue;
-                ObsPoint point = obs.getObsPoint();
-                if (point == null) continue;
-
-                mDays.add(new DateTime(point.time).toLocalDate());
                 Column column = getColumnContainingTime(point.time);
```

Comment:
Fixed now and repushed (which updated this PR).  After removing these four lines, if you replace `point.time` with `obs.time` it works out.